src/main/java/com/google/sps/servlets/AddSchoolInfo.java

```diff
@@ -0,0 +1,159 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import java.io.IOException;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/add-info")
+public class AddSchoolInfo extends HttpServlet {
+
+  private Key existingSchoolKey;
+  private Key existingCourseKey;
+  private Key existingProfessorKey;
+
+  private Key newSchoolKey;
+  private Key newCourseKey;
+  private Key newProfessorKey;
+  private Key newTermKey;
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    String schoolName = request.getParameter("school-name");
+    String courseName = request.getParameter("course-name");
+    String term = request.getParameter("term");
+    Integer units = Integer.parseInt(request.getParameter("units"));
+    String profName = request.getParameter("professor-name");
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+    Filter schoolFilter = new FilterPredicate("school-name", FilterOperator.EQUAL, schoolName);
+    Query schoolQ = new Query("school").setFilter(schoolFilter);
+    List<Entity> schoolQuery =
+        datastore.prepare(schoolQ).asList(FetchOptions.Builder.withDefaults());
+
+    if (schoolQuery.isEmpty()) {
+      Entity newSchool = new Entity("school");
+      newSchool.setProperty("school-name", schoolName);
+      newSchoolKey = newSchool.getKey();
+      datastore.put(newSchool);
+    } else {
+      Entity existingSchool = schoolQuery.get(0);
+      existingSchoolKey = existingSchool.getKey();
+      newSchoolKey = null;
+    }
+
+    if (newSchoolKey != null) {
+      Entity newCourse = new Entity("course", newSchoolKey);
+      newCourse.setProperty("course-name", courseName);
+      newCourse.setProperty("units", units);
+      newCourseKey = newCourse.getKey();
+      datastore.put(newCourse);
+
+      Entity newProfessor = new Entity("professor", newSchoolKey);
+      newProfessor.setProperty("professor-name", profName);
+      newProfessorKey = newProfessor.getKey();
+      datastore.put(newProfessor);
+
+      Entity newTerm = new Entity("term", newCourseKey);
+      newTerm.setProperty("term", term);
+      newTerm.setProperty("professor", newProfessorKey);
+      datastore.put(newTerm);
+
```

Comment:
Nit: extra line? Feel free to ignore this if this is in the style guide - in that case my bad haha

Reply:
Oh I use an extra line to separate each entity. I think its allowed, kinda a grey area

Reply:
Agree with Dagmawi on this, makes it a lot easier to read :)

Reply:
Sounds good :)

Reply:
I also prefer the separate line, clarity is most important here. The main thing we don't like it two separate lines, or a separate line after opening a block via {